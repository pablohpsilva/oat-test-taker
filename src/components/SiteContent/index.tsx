import React, { FC } from 'react'
import styled from 'styled-components/macro'
import useBreakpoint from '@w11r/use-breakpoint'

import ISiteContent from './types'

const StyledSiteContent = styled.div`
    display: flex;
    justify-content: center;
    width: 100%;
    max-width: 100vw;
    min-height: calc(100vh - var(--dim-header-height));
`

const SiteContent: FC<ISiteContent> = ({ children }) => {
    const cols = useBreakpoint('auto auto auto', ['tablet-', 'auto auto'])
    return (
        <StyledSiteContent cols={cols} rows="1fr">
            {children}
        </StyledSiteContent>
    )
}

export default SiteContent
