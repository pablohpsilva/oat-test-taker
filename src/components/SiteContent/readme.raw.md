# SiteContent

This component is reponsible for having a flex wrapper
around the entire site/app content. It is used to
correctly place the content on screen without 
any unpleasants scrolls. It considers the
navbar on its calculations.

## Props

| Property | Description | Type | Default |
| -------- | ----------- | ---- | ------- |
| `-`      | -           | `-`  | `-`     |
