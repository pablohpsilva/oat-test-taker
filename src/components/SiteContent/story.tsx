import React from 'react'
import { storiesOf } from '@storybook/react'

import { SiteContent } from '@/components'
import readme from './readme.raw.md'

const SiteContentComp = () => <SiteContent>This is a content :)</SiteContent>

const opts = {
    notes: { markdown: readme }
}

storiesOf('Components|SiteContent', module)
    .addDecorator(storyFn => <div>{storyFn()}</div>)
    .add('Example', () => <SiteContentComp />, opts)
