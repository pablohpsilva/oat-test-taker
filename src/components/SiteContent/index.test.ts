import React from 'react'
import SiteContent from './'

test('should render the SiteContent component', async () => {
    const wrapper = shallow(
        <div className="SiteContent">
            <SiteContent />
        </div>
    )

    expect(wrapper.first().hasClass('SiteContent')).toMatchSnapshot()
    expect(wrapper).toMatchSnapshot()
})
