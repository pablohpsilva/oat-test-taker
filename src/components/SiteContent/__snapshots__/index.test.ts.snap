// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the SiteContent component 1`] = `true`;

exports[`should render the SiteContent component 2`] = `
<div
  className="SiteContent"
>
  <SiteContent />
</div>
`;
